Sodium/GenericHash.swift

```diff
@@ -147,6 +147,10 @@ public class GenericHash {
             self.outputLength = outputLength
         }
 
+        private func free() {
+            Stream.free(state)
```

Comment:
In its current form, `StateStream` can be implemented privately (inside a `private extension Stream: StateStream`), so that all the members that it prescribes could be private without the workaround of implementing a wrapper `free()`. If done this way, the changes of this commit are unnecessary and the code would IMHO be clearer.

Reply:
I don't think that declaring an accessibility level on conformance is possible – doesn't it have to be the same visibility as the protocol?

Reply:
You're right, I just tried it out. This does only work for extensions that don't declare a protocol conformance.